src/azul/service/async_manifest_service.py

```diff
@@ -63,40 +78,76 @@ class AsyncManifestService(AbstractService):
     """
     step_function_helper = StepFunctionHelper()
 
+    step_function_execution_retries = 3
+
     def __init__(self, state_machine_name):
         self.state_machine_name = state_machine_name
 
     def start_generation(self, input: JSON) -> Token:
-        execution_id = str(uuid.uuid4())
-        self.step_function_helper.start_execution(self.state_machine_name,
-                                                  execution_id,
-                                                  execution_input=input)
-        return Token(execution_id=execution_id,
-                     request_index=0,
-                     wait_time=self._get_next_wait_time(0))
 
-    def inspect_generation(self, token) -> Union[Token, JSON]:
-        try:
-            execution = self.step_function_helper.describe_execution(state_machine_name=self.state_machine_name,
-                                                                     execution_name=token.execution_id)
-        except ClientError as e:
-            if e.response['Error']['Code'] == 'ExecutionDoesNotExist':
-                raise InvalidTokenError from e
+        execution_id = input['object_key'].rpartition('/')[2]
```

Comment:
I think there's an oddity here: The makeup of the object key is not a concern of this service, nevertheless it seems to have intimate knowledge about it.